server/src/test/java/org/elasticsearch/common/geo/CentroidCalculatorTests.java

```diff
@@ -44,12 +52,82 @@ public void test() {
             System.arraycopy(y, 0, subY, 0, i + 1);
             Geometry geometry  = new Line(subX, subY);
             calculator = new CentroidCalculator(geometry);
-            assertThat(calculator.getX(), equalTo(xRunningAvg[i]));
-            assertThat(calculator.getY(), equalTo(yRunningAvg[i]));
+            assertEquals(xRunningAvg[i], calculator.getX(), DELTA);
+            assertEquals(yRunningAvg[i], calculator.getY(), DELTA);
         }
         CentroidCalculator otherCalculator = new CentroidCalculator(new Point(0, 0));
         calculator.addFrom(otherCalculator);
-        assertThat(calculator.getX(), equalTo(50.0));
-        assertThat(calculator.getY(), equalTo(5.0));
+        assertEquals(55.0, calculator.getX(), DELTA);
+        assertEquals(5.5, calculator.getY(), DELTA);
+    }
+
+    public void testRoundingErrorAndNormalization() {
```

Comment:
This is one issue I am not entirely sure how to best support. Due to how the centroid is calculated, it can happen that the longitude or latitude values are invalid and must be normalized.

Reply:
I think this is ok. Normalization only happens when writing the centroid into the doc value.